src/rgw/store/sfs/user.cc

```diff
@@ -133,8 +133,10 @@ int SFSUser::list_buckets(const DoutPrefixProvider *dpp,
 
   std::list<sfs::BucketRef> lst = store->bucket_list();
   for (const auto &bucketref : lst) {
-    auto bucket = std::unique_ptr<Bucket>(new SFSBucket{store, bucketref, bucketref->to_rgw_bucket_info()});
-    buckets.add(std::move(bucket));
+    if (!bucketref->get_deleted_flag()) {
+      auto bucket = std::unique_ptr<Bucket>(new SFSBucket{store, bucketref, bucketref->to_rgw_bucket_info()});
+      buckets.add(std::move(bucket));
```

Comment:
cannot be the `std::move` avoided not declaring the `lval` `bucket` variable (and just constructing the unique ptr in the `add` directly as `rval`) ?
Generally speaking, I dislike explicitly move an `lval` because it introduces a potential flaw in the code (because the user still can access the variable after it has been moved)

Reply:
That was not really part of the PR (just changed due the indentiation) but I've changed it.

Reply:
my comments are not mandatory, just suggestions ;)